python/az/aro/azext_aro/_rbac.py

```diff
@@ -48,5 +46,52 @@ def assign_contributor_to_vnet(cli_ctx, vnet, object_id):
     ))
 
 
+def assign_contributor_to_routetable(cli_ctx, master_subnet, worker_subnet, object_id):
+    auth_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_AUTHORIZATION)
+    client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_NETWORK)
+
+    role_definition_id = resource_id(
+        subscription=get_subscription_id(cli_ctx),
+        namespace='Microsoft.Authorization',
+        type='roleDefinitions',
+        name=DEVELOPMENT_CONTRIBUTOR if rp_mode_development() else CONTRIBUTOR,
+    )
+
+    route_tables = {}
```

Comment:
can you make this a `set()`?  It makes the intention slightly clearer.

Reply:
I actually implemented this change as you previously suggested and it broke the pull request as commented [here](https://github.com/Azure/ARO-RP/pull/728#issuecomment-640679315).  I moved this from a list to a set and now to a dict so that we could use the route table's ID as the key.  This should provide the same functionality as a `set()`.

Reply:
ah, I think you need this:
```
route_tables = set()
...
route_tables.add(subnet.route_table.id)
...

just add the id, which should be hashable.